src/analyser.cpp

```diff
@@ -198,6 +199,129 @@ bool AnalyserInternalEquation::variableOnLhsOrRhs(const AnalyserInternalVariable
            || variableOnRhs(variable);
 }
 
+SymEngine::RCP<const SymEngine::Basic> AnalyserInternalEquation::getSymEngineRepresentation(AnalyserEquationAstPtr ast, const std::map<std::string, SymEngine::RCP<const SymEngine::Symbol>> &symbolMap)
+{
+    if (ast == nullptr) {
+        return SymEngine::null;
+    }
+
+    AnalyserEquationAstPtr leftAst = ast->leftChild();
+    AnalyserEquationAstPtr rightAst = ast->rightChild();
+
+    // Recursively call getConvertedAst on left and right children
+    SymEngine::RCP<const SymEngine::Basic> left = getSymEngineRepresentation(leftAst, symbolMap);
+    SymEngine::RCP<const SymEngine::Basic> right = getSymEngineRepresentation(rightAst, symbolMap);
+
+    // Analyse mAst current type and value
+    switch (ast->type()) {
+    case AnalyserEquationAst::Type::EQUALITY:
+        return Eq(left, right);
+    case AnalyserEquationAst::Type::PLUS:
+        return add(left, right);
+    case AnalyserEquationAst::Type::CI:
+        // Seems like the voi doesn't exist in mAllVariables, so we don't have an easy means of access
+        // For now we'll just throw an error if the symbol is not found
+        if (symbolMap.find(ast->variable()->name()) == symbolMap.end()) {
+            throw std::runtime_error("Unsupported variable in getSymEngineRepresentation");
```

Comment:
libCellML doesn't throw exceptions. So, you need to find another solution. Anyway, at some point, you will also need to account for the VOI (for which a test should be created).